### DISCUSSION TITLE
Aboud black mask image when process local custom data

### DISCUSSION THREAD

**Discussion by MaybeShewill-CV** [MaybeShewill-CV/lanenet-lane-detection#561] [Show and tell]
Lots of issues mentioned about the black mask image problem during process their own custorm data. The problem is mainly caused by the cluster postprocess  strategy. Here is a sample collected in the closed issues(https://github.com/MaybeShewill-CV/lanenet-lane-detection/issues/382). User have found black mask image when using pretrained model to inference his own image data. The test results are listed here.
[mask_image_before_adjust](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/master/data/source_image/black_mask.png)

You may find the instance segmentation result may somehow identify different lanes although it was not that good here but it's a another question. Why would the mask image still become empty? The reason was the dbscan cluster params was not proper configured for that data. When I enlarge the dbscan eps param from 0.35 to 0.5 and reduce min_samples params from 1000 to 250. The cluster could work well on this data. The test was listed here.
[mask_image_after_adjust](https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/master/data/source_image/black_mask_after_adjust.png)

So first thing you may need to do when faced with such a problem is to adjust the dbscan params to see if it helps. But if you can not even identify different lanes on instance segmentation result or binary segmentation result what you need is to retrain the model rather than adjust some of the params.

Anyway hope it helps you guys !!!

**Comment by arijit-imerit** [comment]
Hi, I have downloaded the contents of the Dropbox link and extracted it in a directory named weights/ 
After that I run this command: python3 test_lanenet.py --weights_path weights/ --image_path ~/Downloads/sample.jpg

I am getting the following error:

File "/home/imerit/.local/lib/python3.8/site-packages/tensorflow/python/training/saver.py", line 1290, in restore
    raise ValueError("The passed save_path is not a valid checkpoint: " +
ValueError: The passed save_path is not a valid checkpoint: weights/

Can you kindly help me out?

**Comment by MaybeShewill-CV** [comment]
@arijit-imerit --weights ./weights/xxxx.ckpt

**Reply by arijit-imerit** [reply]
Thank you for replying!
Unfortunately, the Dropbox link mentioned in the ReadME doesn't contain a .ckpt file. There's a checkpoint, a.meta file, a .index file. The other link for BaiduNetDisk isn't working. Would you kindly be able to upload the .ckpt file in the Dropbox?

**Reply by yxgi5** [reply]
in file tools/test_lanenet.py

add
```
import sys
sys.path.append('/ext_drv/mov/lanenet-lane-detection')   # 
```

change
```
    with sess.as_default():
        saver.restore(sess=sess, save_path=weights_path)
```
to
```
    with sess.as_default():
        module_file = tf.compat.v1.train.get_checkpoint_state(weights_path)
        saver.restore(sess, module_file.model_checkpoint_path)
```


change checkpoint file
```
model_checkpoint_path: "tusimple_lanenet.ckpt"
all_model_checkpoint_paths: "tusimple_lanenet.ckpt"
```

run test like
```
python tools/test_lanenet.py --weights_path ./weights/tusimple_lanenet/ --image_path ./data/tusimple_test_image/0.jpg
```

**Comment by MaybeShewill-CV** [comment]
@arijit-imerit You may google how to restore tensorflow model from checkpoint file:)
